internal/glrepo/resolver.go

```diff
@@ -141,6 +141,84 @@ func (r *ResolvedRemotes) BaseRepo(prompt bool) (Interface, error) {
 	return selectedRepoInfo, err
 }
 
+func (r *ResolvedRemotes) HeadRepo(prompt bool) (Interface, error) {
+	if r.baseOverride != nil {
+		return r.baseOverride, nil
+	}
+
+	// if any of the remotes already has a resolution, respect that
+	for _, r := range r.remotes {
+		if r.Resolved == "head" {
+			return r, nil
+		}
+	}
+
+	if !prompt {
+		// we cannot prompt, so just resort to the 1st remote
+		return r.remotes[0], nil
+	}
+
+	// from here on, consult the API
+	if r.network == nil {
+		err := resolveNetwork(r)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var repoNames []string
+	repoMap := map[string]*gitlab.Project{}
+	add := func(r *gitlab.Project) {
+		fn, _ := FullNameFromURL(r.HTTPURLToRepo)
+		if _, ok := repoMap[fn]; !ok {
+			// This is run inside a for-loop, create a local copy and use its address
+			// instead of the one given to us, otherwise the value in repoMap will be
+			// overwriten in the next iteration of the loop
+			// See: #395, #384
+			localCopy := *r
+			repoMap[fn] = &localCopy
+			repoNames = append(repoNames, fn)
+		}
+	}
+
+	for _, repo := range r.network {
+		if repo.ForkedFromProject != nil {
+			fProject, _ := api.GetProject(r.apiClient, repo.ForkedFromProject.PathWithNamespace)
+			add(fProject)
+		}
+		add(&repo)
+	}
+
+	if len(repoNames) == 0 {
+		return r.remotes[0], nil
+	}
+
+	baseName := repoNames[0]
+	if len(repoNames) > 1 {
+		err := survey.AskOne(&survey.Select{
+			Message: "Which should be the base repository (used pushing branches) for this directory?",
```

Comment:
I think the message here conflicts with the prompt message of the base repository. We can use `head`
```suggestion
			Message: "Which should be the head repository (used pushing branches) for this directory?",
```

Reply:
uuugh, this is another change that was lost on the commits I dropped while rebasing.